R/CibersortX.R

```diff
@@ -54,24 +57,33 @@ build_model_cibersortx <- function(single_cell_object, cell_type_annotations,
   if (is.null(cell_type_annotations)) {
     stop("Parameter 'cell_type_annotations' is missing or null, but it is required.")
   }
-  if (!docker_available()) {
-    message(
-      "Installation of docker can not be found. Please check whether you can ",
-      "call 'docker' in the command line and get a help menu"
-    )
-    return(NULL)
-  }
-  if (!docker_connectable()) {
-    message(
-      "Error durching connection to docker. Please check whether you can ",
-      "call 'docker ps' in the command line and get a (possibly empty) list and not an error ",
-      "message"
-    )
-    return(NULL)
+
+  if (container == "docker"){
+    if (!docker_available()) {
+      message(
+        "Installation of docker can not be found. Please check whether you can ",
+        "call 'docker' in the command line and get a help menu"
+      )
+      return(NULL)
+    }
+    if (!docker_connectable()) {
+      message(
+        "Error durching connection to docker. Please check whether you can ",
+        "call 'docker ps' in the command line and get a (possibly empty) list and not an error ",
+        "message"
+      )
+      return(NULL)
+    }
   }
+
   check_credentials()
 
   temp_dir <- tempdir()
+
+  if(container == "singularity"){
+    system(paste0('singularity pull --dir ', temp_dir, ' docker://cibersortx/fractions'))
+  }
```

Comment:
This now creates a new sif file every single time the cibersortx is executed? 
It would be great to find a way to make the `sif` file persistent. 

One option: 
 * add an optional argument `singularity_image="./cibersortx_fractions.sif"` to the function
 * when the file is not present, execute singularity pull
 * if it is present, just use the file

This also allows to use a custom sif file, or customize the path where to cache the container.

Reply:
Ok for the path option. But do we really want the user to use a custom `.sif` file? IMO this could open up the possibility for bugs and incompatibilities since there are so many steps performed on the results

Reply:
We shouldn't encourage them to do so (i.e. per default it should be generated automatically). But I don't think there's anything wrong with having the possibility to provide a custom file (at their own risk, of course)

Reply:
Ok. 
Then I'll do something like this: 

Users can provide an _optional_ path parameter. If in that path there is a `sif` file, then it will be selected. Otherwise, the singularity cibersortx container will be pulled and stored. 
If the path parameter is null, then the `sif` will be pulled and stored in another directory (don't know exactly where to store that since we can not write in the libPath but I'll manage it)

Reply:
you could write in the current working directory. Or into `.local/share/omnideconv` which is somewhat standardized for this purpose: https://askubuntu.com/questions/14535/whats-the-local-folder-for-in-my-home-directory

Reply:
in any case I would generate a log message when `path` is not set, such as
```
singularity container written to `/home/sturm/cibersortx_fractions.sif`. Set the `path` directory to choose a different location
```

Reply:
Solved in [2c37fab](https://github.com/omnideconv/omnideconv/pull/115/commits/2c37fabd4938167641f3e3d31c3e4c6f0484c768)